multiapi-engine/src/main/java/com/sngular/api/generator/plugin/asyncapi/template/TemplateFactory.java

```diff
@@ -368,4 +420,15 @@ private void writeTemplateToFile(final String templateName, final Map<String, Ob
     template.process(root, writer);
     writer.close();
   }
+
+  private String checkTemplate(final String bindingType, final String defaultTemplate) {
+    switch (BindingTypeEnum.valueOf(bindingType)) {
+      case NONBINDING:
+        return defaultTemplate;
+      case KAFKA:
+        return MapperUtil.splitName(defaultTemplate)[0] + "WithKafkaBindings.ftlh";
```

Comment:
Maybe include the template `WithKafkaBindings.ftlh` into the `TemplateIndexConstants.java`as the others templates